### DISCUSSION TITLE
VK Video is not downloading

### DISCUSSION THREAD

**Discussion by vladius1972** [aclap-dev/video-downloadhelper#147] [Help and Q&A]
### Your question

After selecting the download option, the following errors pop up:

SideDownload error:

https://vk9-10.vkuser.net/?srcIp=93.92.202.60&pr=40&expires=1708620270827&srcAg=GECKO&fromCache=1&ms=185.32.248.169&type=4&sig=mZ6e5O60AvY&ct=6&urls=45.136.22.139%3B185.226.52.209&clientType=14&appId=512000384397&zs=12&id=2459330021913:-1: error : unterminated entity reference srcIp=93.92.202.60&pr=40&srcAg=GECKO&ms=185.32.248.169&type=4&sig=mZ6e5O60AvY&ct=21&urls=45.136.22.139%3B185.226.52.209&clientType=14&appId=512000384397&zs=12&id=2459330021913
https://vk9-10.vkuser.net/?srcIp=93.92.202.60&pr=40&expires=1708620270827&srcAg=GECKO&fromCache=1&ms=185.32.248.169&type=4&sig=mZ6e5O60AvY&ct=6&urls=45.136.22.139%3B185.226.52.209&clientType=14&appId=512000384397&zs=12&id=2459330021913:-1: error : unterminated entity reference 
................
https://vk9-10.vkuser.net/?srcIp=93.92.202.60&pr=40&expires=1708620270827&srcAg=GECKO&fromCache=1&ms=185.32.248.169&type=4&sig=mZ6e5O60AvY&ct=6&urls=45.136.22.139%3B185.226.52.209&clientType=14&appId=512000384397&zs=12&id=2459330021913:-1: error : unterminated entity reference srcIp=93.92.202.60&pr=40&srcAg=GECKO&ms=185.32.248.169&type=1&sig=iUVXyQgA-AQ&ct=22&urls=45.136.22.139%3B185.226.52.209&clientType=14&appId=512000384397&zs=12&id=2459330021913
[dash @ 0x5297c80] Error when loading first fragment of playlist
[in#0 @ 0x5297ac0] Error opening input: Invalid data found when processing input
Error opening input file https://vk9-10.vkuser.net/?srcIp=93.92.202.60&pr=40&expires=1708620270827&srcAg=GECKO&fromCache=1&ms=185.32.248.169&type=4&sig=mZ6e5O60AvY&ct=6&urls=45.136.22.139%3B185.226.52.209&clientType=14&appId=512000384397&zs=12&id=2459330021913.
Error opening input files: Invalid data found when processing input

### Addon & CoApp version and details

version: 8.2.0.23
channel: stable
build date: 2024-02-14
build options: linuxlic: false, noyt: false, browser: firefox
lang: ru
license: unneeded
platform: x86-64 linux
UA: Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:122.0) Gecko/20100101 Firefox/122.0
coapp: yes. Info:
{
  "id": "net.downloadhelper.coapp",
  "name": "VdhCoApp",
  "version": "2.0.17",
  "binary": "/opt/vdhcoapp/vdhcoapp",
  "displayName": "VdhCoApp",
  "description": "Video DownloadHelper companion app",
  "target": {
    "os": "linux",
    "arch": "x86_64",
    "node": "18"
  },
  "home": "/home/vlad",
  "converterBinary": "/opt/vdhcoapp/ffmpeg",
  "converterBase": "ffmpeg",
  "converterBaseVersion": "63b936e"
}

**Comment by Mattmjs** [comment]
The dev is still to investigate it, similar to https://github.com/aclap-dev/video-downloadhelper/discussions/101#discussioncomment-8487938

If you don't want to wait, use my tip for VK video.

**Comment by Mattmjs** [comment]
See my comment --->  https://github.com/aclap-dev/video-downloadhelper/discussions/143#discussioncomment-8504200

**Comment by paulrouget** [comment]
I believe the latest coapp (2.0.19) should fix the issue: https://www.downloadhelper.net/w/CoApp-Installation

**Comment by paulrouget** [comment]
For anyone curious, that's how I fixed it: https://trac.ffmpeg.org/ticket/7395#comment:5

**Comment by Mattmjs** [comment]
I'm not a coder / programmer , but you solved a 5 year old bug in ffmpeg.
It was an  escaping issue for special characters if I read that correctly.

It must have been easy as pie to come up with the fix. Does that also mean
it could be utilised in regular builds of ffmpeg.

**Reply by paulrouget** [reply]
I hope this will be included in ffmpeg eventually.

That's why we have to package our own version of ffmpeg. We have plenty of fixes like this.
